feat: Kubernetes Auth Method (#4775)
Kong/docs.konghq.com: src/gateway/reference/configuration.md

> ### keyring_vault_auth_method

Reviewer: if you haven't done it already, this doc is generated by editing this file in `kong-ee` https://github.com/Kong/kong-ee/blob/master/kong.conf.default

Author: I have now done that.

### keyring_vault_kube_role
{:.badge .enterprise}

> Defines the Hashicorp Vault role that will be assumed using the Kubernetes service

Reviewer: ```suggestion
Defines the HashiCorp Vault role for the Kubernetes service
```

account of the running pod.

`keyring_vault_auth_method` must be set to `kubernetes` for this to activate.

**Default:** `default`

### keyring_vault_kube_api_token_file
{:.badge .enterprise}

> Defines where the Kubernetes service account token should be read from the pod's

Reviewer: ```suggestion
Defines where the Kubernetes service account token should be read from on the pod's
```
